custom/enikshay/tasks.py

```diff
@@ -795,28 +813,26 @@ def calculate_dose_status_by_day(adherence_cases):
     adherence_cases: list of 'adherence' case dicts that come from elasticsearch
     Returns: {day: DoseStatus(taken, missed, unknown, source)}
     """
-
-    adherence_cases_by_date = defaultdict(list)
-    for case in adherence_cases:
-        adherence_date = parse_date(case['adherence_date']) or parse_datetime(case['adherence_date']).date()
-        adherence_cases_by_date[adherence_date].append(case)
-
-    status_by_day = defaultdict(lambda: DoseStatus(taken=False, missed=False, unknown=True, source=False))
+    adherence_cases_by_date = get_adherence_cases_by_date(adherence_cases)
+    status_by_day = defaultdict(lambda: DoseStatus(taken=False, missed=False, unknown=True,
+                                                   source=False, value=None))
     for day, cases in six.iteritems(adherence_cases_by_date):
-        case = _get_relevent_case(cases)
+        case = get_relevent_case(cases)
         if not case:
             pass  # unknown
         elif case.get('adherence_value') in DOSE_TAKEN_INDICATORS:
             source = case.get('adherence_report_source') or case.get('adherence_source')
-            status_by_day[day] = DoseStatus(taken=True, missed=False, unknown=False, source=source)
+            status_by_day[day] = DoseStatus(taken=True, missed=False, unknown=False, source=source,
+                                            value=case.get('adherence_value'))
         elif case.get('adherence_value') == DOSE_MISSED:
-            status_by_day[day] = DoseStatus(taken=False, missed=True, unknown=False, source=False)
+            status_by_day[day] = DoseStatus(taken=False, missed=True, unknown=False, source=False,
+                                            value=case.get('adherence_value'))
         else:
             pass  # unknown
     return status_by_day
 
 
-def _get_relevent_case(cases):
+def get_relevent_case(cases):
```

Comment:
I'd rename this now that it is used outside of just this file. There's also a spelling mistake in this name. 
`get_primary_adherence_case` or something similar would be more appropriate.